File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
def isPrime(x):
    if x == 1:
        return False
    for i in range(2, int(x**0.5 + 1)):
        if (x % i == 0):
            return False
    return True

def solution(n, k):
    answer = 0
    num = 0
    while n > 0:
        acc = 1
        tmp = 1
        while (n >= tmp * k):
            tmp *= k
            acc *= 10
        n -= tmp
        num += acc
    tmp = str(num)
    candidates = tmp.replace('0', ' ').split()
    for cand in candidates:
        if isPrime(int(cand)):
            answer += 1
    return answer

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

bool isPrime(long long target) {
    if (target == 1) return false;
    for (long long i = 2; i * i <= target; i++) {
        if (target % i == 0) return false;
    }
    return true;
}

int solution(int n, int k) {
    int answer = 0;
    
    vector<int> target;
    target.resize(20, 0);
    while (n > 0) {
        int power = 0;
        int tmp = 1;
        while (n >= tmp * k) {
            power++;
            tmp *= k;
        }
        n -= tmp;
        target[power]++;
    }
    
    string targetString;
    for (int i = 0; i < (int)target.size(); i++) {
        int curDigit = target[i];
        targetString.insert(0, to_string(curDigit));
    }
    
    int len = 0;
    vector<pair<int, int>> candidates;
    for (int i = 0; i < targetString.length(); i++) {
        if (targetString[i] == '0') {
            if (len) {
                candidates.emplace_back(i - len, len);
                len = 0;
            }
        } else {
            len++;
        }
    }
    if (len) {
        candidates.emplace_back(targetString.length() - len, len);
    }
    
    for (int i = 0; i < candidates.size(); i++) {
        int targetIndex = candidates[i].first;
        int length = candidates[i].second;
        string tmp = targetString.substr(targetIndex, length);
        if (isPrime(stoll(tmp))) answer++;
    }
    
    return answer;
}